lib/block-supports/layout.php

```diff
@@ -572,15 +570,28 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 		);
 
 		$outer_class_names[] = $container_content_class;
+	}
+
+	// Prep the processor for modifying the block output.
+	$processor = new WP_HTML_Tag_Processor( $block_content );
 
+	// Having no tags implies there are no tags onto which to add class names.
+	if ( ! $processor->next_tag() ) {
+		return $block_content;
 	}
 
-	// Return early if only child layout exists.
+	/*
+	 * Return early if only child layout exists.
+	 *
+	 * In other words, if there is nothing more complicated, add
```

Comment:
Can you explain the original comment in different words? I think my ambiguity here comes from my inability to understand what "if only child layout exists" means.

Reply:
specifically, what does `has_child_layout` mean?

Reply:
Hmm yeah I could have done a better job with the original comment 😅 

`has_child_layout` means that the block is inside a block with flex layout that has `allowSizingOnChildren` set to true, so it has access to certain layout properties, even if the block doesn't itself support layout.

Currently the only properties applied to flex children are `selfStretch` and `flexSize`, but more might be added in the future. Any block at all that is a child of a flex block with `allowSizingOnChildren` (currently the only core blocks that support it are Group variations and Navigation) has access to those properties, through a set of controls in the Dimensions panel:

<img width="284" alt="Screenshot 2023-09-04 at 9 57 49 am" src="https://github.com/WordPress/gutenberg/assets/8096000/33e6eed5-fa0a-4be9-8c18-f35f7bc8b85e">

So at this point in the code we're checking if the block supports layout, because if it doesn't we can just add whatever child layout properties it has and skip all the logic below.

Reply:
thanks for clarifying; I've attempted to explain this in comments in the latest version of the patch.

```php
	/*
	 * A block may not support layout but still be affected by a parent block's layout.
	 *
	 * In these cases add the appropriate class names and then return early; there's
	 * no need to investigate on this block whether additional layout constraints apply.
	 */
```